### PR TITLE
release/public-v1: update and cleanup of build configurations

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

(1) Update modulefiles for hera.intel and jet.intel in accordance with NCEPLIBS-external/NCEPLIBS update. It was necessary to  build our own netCDF libraries as part of NCEPLIBS-external, because using the default modules led to shared library issues with libsz in the ncep_post executable.

(2) Remove modulefiles, gnumake and cmake configurations for all platforms except level-1 (preconfigured). This is to avoid confusion, because the modulefiles for level 2/3 platforms by definition cannot contain libraries in shared spaces (libraries for the UFS MRW App are installed by the user in his/her own space).

### Manual testing on pre-configured, configurable and limited testing platforms

- ufs-weather-model compiles using `build.sh` and `compile_cmake.sh` on the following pre-configured platforms: hera.intel, cheyenne.intel, cheyenne.gnu, gaea.intel, jet.intel
- ufs-weather-model compiles on tier-2 platforms (orion, stampede) using `build.sh`
- ufs-weather-model compiles on macOS with clang+gfortran and gcc+gfortran using `build.sh`
- ufs-weather-model compiles on Ubuntu 18 and RedHat 8 Amazon cloud instances using `build.sh`

### Regression testing on cheyenne.intel, cheyenne.gnu, hera.intel

Regression tests against existing baselines:
- all tests pass on cheyenne.gnu and cheyenne.intel
[rt_cheyenne_gnu.log](https://github.com/ufs-community/ufs-weather-model/files/5141741/rt_cheyenne_gnu.log)
[rt_cheyenne_intel.log](https://github.com/ufs-community/ufs-weather-model/files/5141742/rt_cheyenne_intel.log)
- on hera.intel, all tests fail due to netCDF/HDF5 metadata mismatches (caused by the HDF5+netCDF library update) in `atmos_4xdaily*.nc` and `RESTART/*.nc` files; however, for all of the tests, the `phyf*.nc` and `dynf*.nc` files are b4b identical
[rt_hera_intel.log](https://github.com/ufs-community/ufs-weather-model/files/5141744/rt_hera_intel.log)
[rt_hera_intel_fail_test.log](https://github.com/ufs-community/ufs-weather-model/files/5141745/rt_hera_intel_fail_test.log)
[log_hera.intel.tar.gz](https://github.com/ufs-community/ufs-weather-model/files/5141747/log_hera.intel.tar.gz)

Creating new baselines `ufs-public-release-20200728`:
- for cheyenne.{intel,gnu}, I simply copied `ufs-public-release-20200618` to `ufs-public-release-20200728`
- on hera.intel: done

Final regression testing against new baselines:
- on cheyenne.{intel,gnu}, all tests pass; logs updated in the PR
[rt_cheyenne_gnu.log](https://github.com/ufs-community/ufs-weather-model/files/5142234/rt_cheyenne_gnu.log)
[rt_cheyenne_intel.log](https://github.com/ufs-community/ufs-weather-model/files/5142236/rt_cheyenne_intel.log)
- on hera.intel, all tests pass; logs updated in the PR
[rt_hera_intel.log](https://github.com/ufs-community/ufs-weather-model/files/5143111/rt_hera_intel.log)

## Dependencies

- waiting on https://github.com/NOAA-EMC/fv3atm/pull/163

### CODE REVIEW

**Comment by uturuncoglu** [comment]
@climbfuji @ligiabernardet this PR along with #191 #193 seems changing the build system. It seems NCEPLIBS_DIR is no longer used. If so, this will affect the CIME build that requires additional work and we need to discuss about it. Are we missing something in here? If you could clarify it that would be great.

**Comment by climbfuji** [comment]
> @climbfuji @ligiabernardet this PR along with #191 #193 seems changing the build system. It seems NCEPLIBS_DIR is no longer used. If so, this will affect the CIME build that requires additional work and we need to discuss about it. Are we missing something in here? If you could clarify it that would be great.

NCEPLIBS_DIR is still to be used for the tier-2 and tier-3 platforms, just not for the tier-1 platforms.

**Comment by jedwards4b** [comment]
So this will break the build on tier-1 systems.

**Comment by climbfuji** [comment]
> So this will break the build on tier-1 systems.

No, why? It was decided to load the tier-1 modulefiles from ufs-weather-model instead of generating it from CIME. These modulefiles still set NCEPLIBS_DIR and should work out of the box. They (or, more precisely, the standalone ufs-weather-model build) don't rely on NCEPLIBS_DIR.

**Comment by uturuncoglu** [comment]
@climbfuji as I remember, we did not decide to add that capability or use it, we just discussed about its possibility. We did not even test it and we don't know it is working or not for this particular case. It also creates additional problem with other UFS applications such as HAFS. Currently, we are using same machine file for all UFS apps and the changes need to be test with all of them to be sure it is not breaking anything in other apps. So, that changes might require additional changes in the CIME side and it could have side effects. As I know, we need to keep the changes minimal in this release compared to v1.0. We already make couple of changes in the CIME interface.

**Comment by climbfuji** [comment]
> @climbfuji as I remember, we did not decide to add that capability or use it, we just discussed about its possibility. We did not even test it and we don't know it is working or not for this particular case. It also creates additional problem with other UFS applications such as HAFS. Currently, we are using same machine file for all UFS apps and the changes need to be test with all of them to be sure it is not breaking anything in other apps. So, that changes might require additional changes in the CIME side and it could have side effects. As I know, we need to keep the changes minimal in this release compared to v1.0. We already make couple of changes in the CIME interface.

I have no preference on how CIME is using or generating the modulefiles. If you think it is easier to keep the MRW v1.0 way of generating them, then that is ok with me as long as they are updated and match the ones in ufs-weather-model for the tier-1 platforms. I will also update the `doc/README_*.txt` files in the release/public-v1 branch of NCEPLIBS-external so that they provide the correct information for all (tier 1, tier 2, tier 3) systems. The two files that need to change, as far as I am aware of, are `doc/README_hera_intel.txt` and `doc/README_jet_intel.txt`. I will tag you in the NCEPLIBS-external PR.

**Comment by uturuncoglu** [comment]
@climbfuji Thanks. Yes, I think it is better and easier to be close as much as possible to v1.0 structure to prevent any potential error. Anyway, we could discuss it in the call.

**Comment by climbfuji** [comment]
@junwang-noaa @DusanJovic-NOAA this PR and the fv3atm PR https://github.com/NOAA-EMC/fv3atm/pull/163 are ready to merge.

**Review by DusanJovic-NOAA — APPROVED** [review]

File: modulefiles/hera.intel/fv3 [hunk 1]
```diff
@@ -21,7 +21,6 @@ module load sutils
 ##
 module load intel/18.0.5.274
 module load impi/2018.0.4
-module load netcdf/4.7.0
```
Comment (junwang-noaa) [inline modulefiles/hera.intel/fv3:24]: Do we not need netcdf lib? I am curious why module files on other platforms are removed e.g. wcoss, cheyenne etc?

**Review by climbfuji — COMMENTED** [review]
The release/public-v1 branch needs modulefiles only for pre-configured platforms for the release. These modulefiles have to match what is stored in the CIME machine config. The module files for the other machines (wcoss, ...) were outdated because not used and not tested. Cheyenne is kept, by the way (for intel and gnu).

netCDF is provided by the NCEPLIBS-external install as described in the PR above, to fix linker issues on the NOAA RDHPC platforms after the UPP (`ncep_post`) updates were made.

File: cmake/configure_wcoss_cray.cmake [hunk 2]
```diff
@@ -1,34 +0,0 @@
-message("")
```
Comment (junwang-noaa) [inline cmake/configure_wcoss_cray.cmake:1]: I am curious why wcoss platforms configuration files are removed, aren't they level-1 platforms?

Comment (climbfuji) [reply cmake/configure_wcoss_cray.cmake:1]: No, they are not supported at all for UFS MRW v1.0 or v1.1: https://github.com/ufs-community/ufs/wiki/Supported-Platforms-and-Compilers

**Review by junwang-noaa — APPROVED** [review]

**Pull request merged**